site/topics/tuples-dictionaries-sets/tuples-dictionaries-sets.rst

```diff
@@ -49,122 +49,112 @@ Tuples
-   
-    Figure out how to access the value associated with a particular key, without printing out the whole dictionary (e.g., how would I print just Suzy's grade?). 
+    # Add a few things to the dictionary
+    some_dictionary["Billy"] = 74
+    some_dictionary["Sally"] = 88
+    some_dictionary["Jimmy-Bob"] = 99
 
-    *Hint*: it's a lot like indexing a list or array or tuple...
-   
-    What happens if I try to index the dictionary with a key that doesn't exist?
-   
-* Dictionaries are a *generalization* of lists:
-    * A list associates *fixed indices* from 0 up to ``n`` with values.
-    * A dictionary associates *arbitrary strings* with values.
+    # Print out the dictionary
+    print(some_dictionary)      # Results in {'Billy': 74, 'Sally': 88, 'Jimmy-Bob': 99}
 
-.. admonition:: Activity
-    :class: activity
 
-    Now type ``my_dict.`` and hit the [Tab] key. Play around with the built-in functions for dictionaries. 
+* In the example, a dictionary was created and three values were added to the dictionary
+* But values are associated with unique *keys*
 
-    Take special care to look at: 
+    * The keys must be unique, but the values do not need to be
 
-        * ``my_dict.keys()``
-        * ``my_dict.values()``
+* The keys in the example are ``"Billy"``, ``"Sally"``, and ``"Jimmy-Bob"``
+* Each of the keys have an associated value --- ``74``, ``88``, and ``99`` respectively
 
-    I wonder if there is an easy way to iterate over the contents of a dictionary?
-   
-   
-* This is *really useful* for humans because it's much easier for us to assign names to things than to try to remember arbitrary numberings.
-  
-* Many programming languages have nothing like dictionaries. In some others you'll see them called "associative arrays" or "associative memories".
-    * In some, we have to *make* them ourselves
+* Accessing a value from a specific key from the dictionary is done with indexing
 
-* We've just scratched the surface of what you can do with dictionaries here, but it's enough for our purposes right now.
+.. code-block:: python
+    :linenos:
 
-.. raw:: html
+    print(some_dictionary["Jimmy-Bob"])     # Results in 99
+    print(some_dictionary["Sally"])         # Results in 88
 
-	<iframe width="560" height="315" src="https://www.youtube.com/embed/e8nhfwlsBl0" frameborder="0" allowfullscreen></iframe>
 
-So Why Are They Great?
-----------------------
+* And updating a value associated with a key is done just like the original assignment
 
-* Imagine we have a 2D list like this:
+    * Keys are unique, so using an existing key would overwrite the value and not make a new entry
 
 .. code-block:: python
-   :linenos:
+    :linenos:
+
+    some_dictionary["Sally"] = 90
+    print(some_dictionary["Sally"])         # Results in 90
+
 
-    # Creates some lookup table list thing
-    grades = [['James', 98], ['Bob', 86], ['Janice', 86], ['Greg', 59]]
+Why They Are Great
+------------------
```

Comment:
This hardly does it justice, but I guess it'll suffice. I'm not really sure how to say "maps are _the_ primary data structure" and have that carry any meaning in this context.